metamath-rs/src/grammar.rs

```diff
@@ -1735,9 +1735,14 @@ impl StmtParse {
         println!("Formula Dump:");
         let sset = db.parse_result();
         let nset = db.name_result();
-        for sps in self.segments.values() {
-            for (&sa, formula) in &sps.formulas {
-                let sref = sset.statement(sa);
+        for seg in sset.segments(..) {
+            let Some(sps) = self.segments.get(&seg.id) else {
+                continue;
+            };
+            for sref in seg.range(..) {
+                let Some(formula) = sps.formulas.get(&sref.address()) else {
+                    continue;
+                };
```

Comment:
Shouldn't we `.unwrap()` instead of `else { continue; }` ?
A formula stored in the wrong segment is an internal bug which should not be just ignored IMO.

Reply:
No, I think this one is a continue because `sref` is looping over *all* statements, including comments, $d declarations, and other random junk.